### PR TITLE
[CU-2jtma61] Oracle Price Feed Initialization Script Rework

### PR DESCRIPTION
**Pull request by lilith-roth**
## Issue
https://app.clickup.com/t/2jtma61

## Description
The old code was badly structured in my opinion, was missing key things like a README.md, and wasn't touched in 4 months.

Instead of continuing on the old structure, I reworked the whole scripts & directories.
A lot of code snippets are copied over from the integration tests.

For those reason i also requested a CODEOWNERS change for this directory.

## Checklist

~~- [ ] I have updated the cargo docs to reflect changes made by this PR _(if applicable)_~~
~~- [ ] I have updated the `book/` to reflect changes made by this PR _(if applicable)_~~

### CODE REVIEW

**Comment by ComposableFiB0t** [comment]
Task linked: [CU-2jtma61 Oracle Price Feed Setup Script](https://app.clickup.com/t/2jtma61)

File: scripts/oracle-setup/src/types/interfaces/pablo/index.js.map [hunk 1]
```diff
@@ -0,0 +1 @@
+{"version":3,"file":"index.js","sourceRoot":"ts/","sources":["src/types/interfaces/pablo/index.ts"],"names":[],"mappings":";AAAA,kEAAkE;AAClE,oBAAoB;;;;;;;;;;;;;;;;AAEpB,0CAAwB"}
```
Comment (benluelo) [inline scripts/oracle-setup/src/types/interfaces/pablo/index.js.map:1]: should the JS files be committed?

**Review by benluelo — COMMENTED** [review]
Are the typescript type definitions the same as the ones from the integration tests? if so, is there a way to reference those types instead of having them duplicated?

**Comment by Dom-Roth** [comment]
> Are the typescript type definitions the same as the ones from the integration tests? if so, is there a way to reference those types instead of having them duplicated?

Yes i can check for this. 
Since there's an npm package with types for our chain in development, I didn't want to put too much effort into the types.

File: scripts/oracle-setup/src/types/interfaces/pablo/index.js.map (hunk 1 above)
Comment (Dom-Roth) [reply scripts/oracle-setup/src/types/interfaces/pablo/index.js.map:1]: Actually not! Thanks for noticing. :)

**Comment by benluelo** [comment]
> Since there's an npm package with types for our chain in development, I didn't want to put too much effort into the types.

Oh I didn't know this, that's awesome :raised_hands: Makes sense then :+1:

File: scripts/oracle-setup/src/handlers.ts [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+import { KeyringPair } from "@polkadot/keyring/types";
+
+/**
+ * Establishes connection.
+ *
+ * @returns {
+ * api: ApiPromise,
+ * keyring: Keyring,
+ * walletAlice: KeyringPair,
+ * }
```
Comment (benluelo) [inline scripts/oracle-setup/src/handlers.ts:22]: I don't believe you need the jsdoc comments and the ts types(?) just one should work

File: scripts/oracle-setup/src/handlers.ts [hunk 3]
```diff
@@ -0,0 +1,176 @@
...
+  const api = newClient;
+  const keyring = newKeyring;
+  return { api: api, keyring: keyring };
+}
+
+/**
+ * Creates Oracle for the given asset ID.
+ *
+ * @param {ApiPromise} api Connected API client.
+ * @param {AddressOrPair} wallet KeyringPair object of the wallet to use.
```
Comment (benluelo) [inline scripts/oracle-setup/src/handlers.ts:35]: same as previous comment, the types can (probably?) be removed from @param

**Review by benluelo — DISMISSED** [review]

**Comment by hussein-aitlahcen** [comment]
Can we resolve this conflict? I'am about to approve :)

**Review by KaiserKarel — APPROVED** [review]

**Review by benluelo — APPROVED** [review]

**Comment by benluelo** [comment]
Bors r+

**Pull request merged**